src/components/PopupWithForm.js

```diff
@@ -0,0 +1,41 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithForm extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._formElement = this._popupElement.querySelector(".modal__form");
+    this._inputList = this._formElement.querySelectorAll(".modal__input");
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  
+  _getInputValues() {
+    const formValues = {};
+    this._inputList.forEach(input => {
+      formValues[input.name] = input.value;
+    });
+    return formValues;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._formElement.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleFormSubmit(this._getInputValues());
+    });
+  }
+
+  close() {
+    super.close();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.